src/server/generic_family.cc

```diff
@@ -623,9 +620,20 @@ void OpScan(const OpArgs& op_args, const ScanOpts& scan_opts, uint64_t* cursor,
   PrimeTable::Cursor cur = *cursor;
   auto [prime_table, expire_table] = db_slice.GetTables(op_args.db_cntx.db_index);
   string scratch;
+  size_t buckets_iterated = 0;
+  const size_t limit = 10000 / (PrimeTable::kBucketNum * PrimeTable::kSlotNum);
```

Comment:
what is the rationale behind this formula? why not just an simple number like 1000?

Reply:
Approximation -- roughly how many buckets are required for 10k elements assuming Traverse goes over a single bucket (which is a lie because a logical bucket contains the two probes + stash). I could hardcode it, I do not have a strong opinion on this

Reply:
i do not think it is required - let's measure in "buckets" or Traverse calls.